src/core/users/MongoDBTiersRepository.ts

```diff
@@ -20,11 +20,14 @@ export class MongoDBTiersRepository implements TiersRepository {
     this.collection = mongo.db('payments').collection<Tier>('tiers');
   }
 
-  async findByProductId(productId: Tier['productId']): Promise<Tier | null> {
-    const tier = await this.collection.findOne({
-      productId,
-    });
+  async findByProductId(productId: Tier['productId'], billingType?: Tier['billingType']): Promise<Tier | null> {
```

Comment:
Why not

`async findByProductId(productId: Tier['productId'], where: Partial<Tier>): Promise<Tier | null>`
?

Reply:
So you do not have to do the magic trick inside the function (unknowns, undefined, etc)

Reply:
Much cleaner. Updating it 🚀